### DISCUSSION TITLE
MHI2Q FEC not added or illegal

### DISCUSSION THREAD

**Discussion by Wizzy088** [Mr-MIBonk/M.I.B._More-Incredible-Bash#70] [Q&A] [answered]
Hi all, I can't figure this one out. Unit patched successfully to MHI2Q_ER_AUG22_P5092, VIM activated. Audi Smartphone Interface already present ex-factory. Now I want to update EU maps. FEC 0023000EE is absent (not being copied) if I follow procedure (use AddFec function in M.I.B.). Or, alternatively, FEC is added as illegal if I manually edit addFecs.txt and copy this file to "patches/MHI2Q_ER_AUG22_P5092/addFec" folder. Clean SWDL clears the illegal FEC entries and then it starts all over again. What am I missing? Is my MHI2 unit properly patched in the first place? Many thanks in advance for your help.
[MU1329-JTB-07418.10.1611740D01.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10196521/MU1329-JTB-07418.10.1611740D01.log)

**Comment by harman-f** [comment]
Look like you did not put the patch files on your SD card or placed them at the wrong foler.

https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/wiki/Prepare-M.I.B-SD-card-&-install-M.I.B#add-patch-files

**Reply by Wizzy088** [reply]
Thanks, I do have the patch files on SD. In REM the train version reads MU1329, which is the target version. But whether patched or not, I can't tell. I will try copying the patch files directly to the /patch folder (with only addFec and belonging dir's as subdir) and report back.

**Reply by Wizzy088** [reply]
A little progress: FEC's are now being added without errors. I read the code of the addfec app and put addFecs.txt in the right dir. However: Added FEC's are marked illegal in REM. Running Clean SWDL clears illegal FEC's. And now I am right back where I started: No lifetime maps still :-(. What's next: back to stock FW and run AIO from scratch? Because I think unit is updated, but not patched somehow.

**Comment by harman-f** [comment, marked as answer]
RTFM

https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/wiki/Prepare-M.I.B-SD-card-&-install-M.I.B#%EF%B8%8F-trouble-shooting-%EF%B8%8F

You SD card is bad. 
Luckily, it seems, that ifs-root was not flashed yet --> illeagl FECs

Get a fresh SD card and try again.

**Reply by Wizzy088** [reply]
OK, will do. M.I.B. requires FAT32, thus limited to max 32GB SD card size. However for map updates and others (e.g. Gracenote), I should be able to use exFAT or NTFS and go for 64GB or even 128GB. Right?

**Reply by harman-f** [reply]
You can always partition an SD card or use cmd line in windows to also format a 64GB SD card wit fat32.

**Reply by Wizzy088** [reply]
I tried a fresh SD card, no luck. The patch just will not install, whatever I try. Remember I already updated the FW to latest version, but I did that from REM directly, without M.I.B. Probably running through M.I.B. is what I failed to do and cannot be corrected afterwards? I think only solution would be to flash to stock FW and restart the update procedure. Besides I don't have the stock version anymore, It is too much hassle for me and too risky also. Everything is fine, no errors, SVM error cleared. Just have to pay for any map updates, just as most of us would have to anyway. I call it quits. Many thanks for your support, I really appreciate it.

**Reply by Wizzy088** [reply]
For the record: I finally pulled it off. The ifs root part 2 patch file was missing. Copied it in the right directory, ran M.I.B. AIO once more and voila: success! Everthing is well documented, I found my answer in the Troubleshooting section, "Patch missing?". You guys keep up the good work. The M.I.B. tool is an absolute gem, it's mostly the user who messes up.
